resteasy-rxjava/src/main/java/org/jboss/resteasy/rxjava/ObservableRxInvokerImpl.java

```diff
@@ -0,0 +1,274 @@
+package org.jboss.resteasy.rxjava;
+
+import java.util.List;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.ScheduledExecutorService;
+
+import javax.ws.rs.ProcessingException;
+import javax.ws.rs.client.Entity;
+import javax.ws.rs.client.SyncInvoker;
+import javax.ws.rs.core.GenericType;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.sse.InboundSseEvent;
+import javax.ws.rs.sse.SseEventSource;
+
+import org.jboss.resteasy.client.jaxrs.internal.ClientInvocationBuilder;
+import org.jboss.resteasy.plugins.providers.sse.InboundSseEventImpl;
+import org.jboss.resteasy.plugins.providers.sse.client.SseEventSourceImpl;
+import org.jboss.resteasy.plugins.providers.sse.client.SseEventSourceImpl.SourceBuilder;
+import org.jboss.resteasy.rxjava.i18n.Messages;
+
+import rx.Observable;
+import rx.Subscriber;
+
+/**
+ * @deprecated:
+ * 
+ *   "RxJava 1.x is now officially end-of-life (EOL). No further developments,
+ *    bugfixes, enhancements, javadoc changes or maintenance will be provided by
+ *    this project after version 1.3.8." - From https://github.com/ReactiveX/RxJava/releases
+ *    
+ *    Please upgrade to resteasy-rxjava2 and RxJava 2.x.
+ */
+@SuppressWarnings("deprecation")
+public class ObservableRxInvokerImpl implements ObservableRxInvoker
+{
+	private static Object monitor = new Object();
```

Comment:
Hey @ronsigal,

I was having a look to this class, in relation to what I've done in https://github.com/resteasy/Resteasy/pull/1599, and just for understanding I was wondering what is this `monitor` for?
According to the code I suppose that it prevents both multiple and concurrent invocations of  `sseEventSource.open(...)` for a given eventSource. I'm OK with it.
But does this monitor needs to be shared beetwen all Observable instance ?

Thanks